src/Modules/OrchardCore.Commerce/Handlers/UniqueSkuValidationHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Lombiq.HelpfulLibraries.OrchardCore.Contents;
+using OrchardCore.Commerce.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Handlers;
+using OrchardCore.DisplayManagement.ModelBinding;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace OrchardCore.Commerce.Handlers;
+
+public class UniqueSkuValidationHandler : ContentPartHandler<ProductPart>
+{
+    private readonly ISession _session;
+    private readonly IUpdateModelAccessor _updateModelAccessor;
+
+    public UniqueSkuValidationHandler(
+        IContentManager contentManager,
+        ISession session,
+        IUpdateModelAccessor updateModelAccessor)
+    {
+        _session = session;
+        _updateModelAccessor = updateModelAccessor;
+    }
+
+    public override async Task UpdatedAsync(UpdateContentContext context, ProductPart instance)
+    {
+        var products = (await _session
+            .QueryContentItem(PublicationStatus.Published)
+            .ListAsync()).AsList();
```

Comment:
You are downloading literally every content item in the DB. This will be slow (and possibly timeout) on any decently sized webapp, so please don't do that.

Reply:
My problem is that I have to check the SKU because that has SKU but literally any ContentType there will be on the site using OCC could have ProductPart on it... Is there a way to tell the DB I only want to query those ContentItems that have ProductPart on them? Because I have never done such query before.

Reply:
We already have [`ProductPartindex`](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/main/src/Modules/OrchardCore.Commerce/Indexes/ProductPartIndex.cs). It contains properties for both ContentItem and Sku too. Use `.With<TIndex>()` to filter by it.

Reply:
I am experimenting with writing a Query instead of QueryContentItem I will update here

Reply:
Why? `session.QueryContentItem(PublicationStatus.Published)` is just `session.Query<ContentItem, ContentItemIndex>(index => index.Published)`.

Reply:
`Query<ProductPart, ProductPartIndex>(index => index.Sku == instance.Sku && index.ContentItemId != instance.ContentItem.ContentItemId)` this yields no result although it really should because the criteria is met I am checking in DB browser. I don't know why but if I can't DB side check SKUs I won't be able to omit querying all the contentitems.

Reply:
That's not acceptable, because as I said it will timeout on a bigger site.

`Query<ProductPart, ProductPartIndex>` should be `Query<ContentItem, ProductPartIndex>`.

Reply:
Same goes for 
`Query().For<ProductPart>().With<ProductPartIndex>(index => index.Sku == instance.Sku)` returns null altough there is like 5 contentitems with the given SKU already

Reply:
okay the problem was refering to  ProductPart

Reply:
Again, you are supposed to query ContentItem, not ProductPart.

Reply:
Yes I know now, just your messages were coming with a Delay so I already wrote my question when I saw your answer